android/lib/src/main/java/com/ern/api/impl/navigation/ElectrodeNavigationFragmentDelegate.java

```diff
@@ -0,0 +1,313 @@
+        mNavViewModel.registerNavRequestHandler();
+    }
+
+    @SuppressWarnings("unused")
+    @CallSuper
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        mMenu = menu;
+        updateNavBar(mFragment.getArguments());
+    }
+
+    @SuppressWarnings("unused")
+    @CallSuper
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            if (mFragment.getActivity() != null) {
+                mFragment.getActivity().onBackPressed();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @CallSuper
+    public void onPause() {
+        super.onPause();
+        mNavViewModel.unRegisterNavRequestHandler();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        if (mMenu != null) {
+            mMenu = null;
+        }
+        mFragmentNavigator = null;
+    }
+
+    public void setMenuItemDataProvider(@NonNull MenuItemDataProvider menuItemDataProvider) {
+        mMenuItemDataProvider = menuItemDataProvider;
+    }
+
+    private void back(@NonNull Route route) {
+        //Manage fragment back-stack popping. If the given route.path is not in the stack pop a new fragment.
+        boolean result = mElectrodeActivityListener.backToMiniApp(route.getArguments().getString("path"), route.getArguments());
+        route.setResult(result, !result ? "back navigation failed. component not found in the back stack" : null);
+    }
+
+    private void update(@NonNull Route route) {
+        if (mFragment.getArguments() != null) {
+            mFragment.getArguments().putAll(route.getArguments());
+        }
+        boolean result = updateNavBar(route.getArguments());
+        route.setResult(true, !result ? "failed to update nav bar." : null);
+    }
+
+    private void finish(@NonNull Route route) {
+        Logger.d(TAG, "finish triggered by RN. Hosting activity will be notified.");
+        mElectrodeActivityListener.finishFlow(NavUtils.getPayload(route.getArguments()));
+        route.setResult(true, null);
+    }
+
+    private void navigate(@NonNull Route route) {
+        final String path = NavUtils.getPath(route.getArguments());
+        Logger.d(TAG, "navigating to: " + path);
+
+        if (!TextUtils.isEmpty(path)) {
+            //If the hosting activity or fragment has not handled the navigation fall back to the default.
+            if (!mElectrodeActivityListener.navigate(path, route.getArguments()) && (mFragmentNavigator == null || !mFragmentNavigator.navigate(path, route.getArguments()))) {
+                LaunchConfig launchConfig = createNextLaunchConfig(route);
+                if (mOnUpdateNextPageLaunchConfigListener != null) {
+                    //noinspection ConstantConditions
+                    mOnUpdateNextPageLaunchConfigListener.updateNextPageLaunchConfig(path, launchConfig);
+                }
+                //noinspection ConstantConditions
```

Comment:
Just wondering: What does this mean?

Reply:
AS shows a lint warning here related to `path` being null without accounting the check `if (!TextUtils.isEmpty(path)) {`. This is to ignore the inspection on this line.

Reply:
Cool. Now that the parameter is `@NonNull`, you can remove it? :)